src/analyze/FrameSizeDeterminationPass.cpp

```diff
@@ -108,19 +110,40 @@ class FrameSizeDeterminationVisitor final : public RecursiveVisitor
 
     std::size_t m_numberOfLocals;
     std::size_t m_maxNumberOfLocals;
+    u1 m_foundProgramFunction;
 };
 
 FrameSizeDeterminationVisitor::FrameSizeDeterminationVisitor( libcasm_fe::Logger& log )
 : RecursiveVisitor()
 , m_log( log )
 , m_numberOfLocals( 0 )
 , m_maxNumberOfLocals( 0 )
+, m_foundProgramFunction( false )
 {
 }
 
+void FrameSizeDeterminationVisitor::visit( Specification& node )
+{
+    node.header()->accept( *this );
+    node.definitions()->accept( *this );
+
+    if( not m_foundProgramFunction )
```

Comment:
These changes don't belong here (i.e. have nothing in common with "frame size calculation").

Reply:
yes you are right, I've removed this check and fixed this directly in the `NumericExecutionPass`